src/pipeline/links.ts

```diff
@@ -481,7 +481,7 @@ class SchemaLinkTransformer implements ExtendedSchemaTransformer {
                     return fetchDeferred(key, info, context);
                 }
             },
-            type: isListMode ? new GraphQLList(targetRawType) : targetRawType
+            type: isListMode || linkConfig.oneToMany ? new GraphQLList(targetRawType) : targetRawType
```

Comment:
* add a check that `oneToMany` is not combined with `batchMode` (this would not work)
* add a check that `targetRawType` is a `GraphQLList` (or a non-null thereof)
* add a check in the `join` handler below that the `link` being referred to is not a `oneToMany`

Reply:
I've added all these checks, but when checking for GraphQLList I needed to use `targetField.type`, not `targetRawType` since that one appears to contain the underlying type of the GraphQLList itself.

Reply:
Well that's why it's called *rawType*, my bad 😆  (don't know how to remove the *pending* state here, this is done.)